src/components/ToDoComponent/SettingGoals/SettingGoalsComponents/DurationDropDown.tsx

```diff
@@ -0,0 +1,85 @@
+import { View, Text, TouchableOpacity, FlatList } from 'react-native';
+import React, { useState } from 'react';
+import { SafeAreaView } from 'react-native-safe-area-context';
+const timeperiod = [
+  { names: 'Short term' },
+  { names: 'Mid term' },
+  { names: 'Long term' },
+];
+const DurationDropDown = () => {
+  const [clicked, setClicked] = useState(false);
+  const [data, setData] = useState(timeperiod);
+  const [selectedTimePeriod, setSelectedTimePeriod] = useState('');
+
+  return (
+    <View>
+      <TouchableOpacity
+        style={{
+          width: '100%',
+          height: 50,
+          elevation: 5,
+          borderRadius: 10,
+          backgroundColor: 'white',
+          alignSelf: 'center',
+          marginTop: 10,
+          flexDirection: 'row',
+          justifyContent: 'space-between',
+          alignItems: 'center',
+          paddingLeft: 15,
+          paddingRight: 15,
+        }}
+        onPress={() => {
+          setClicked(!clicked);
+        }}
+      >
+        <Text style={{ fontWeight: '600', color: 'black' }}>
+          {selectedTimePeriod === '' ? 'Select duration' : selectedTimePeriod}
+        </Text>
+        {clicked ? (
+          <Text style={{ color: 'black', fontSize: 20 }}>-</Text>
+        ) : (
+          <Text style={{ color: 'black', fontSize: 20 }}>+</Text>
+        )}
+      </TouchableOpacity>
+      {clicked ? (
+        <SafeAreaView>
+          <FlatList
+            style={{
+              elevation: 5,
+              marginTop: 20,
+              height: 100,
+              alignSelf: 'center',
+              width: '90%',
+              backgroundColor: '#fff',
+              borderRadius: 10,
+            }}
+            data={data}
+            renderItem={({ item }) => {
+              return (
+                <TouchableOpacity
+                  style={{
+                    width: '85%',
+                    alignSelf: 'center',
+                    height: 50,
+                    justifyContent: 'center',
+                    borderBottomWidth: 0.5,
+                  }}
+                  onPress={() => {
+                    setSelectedTimePeriod(item.names);
+                    setClicked(!clicked);
+                  }}
+                >
+                  <Text style={{ fontWeight: '600', color: 'black' }}>
+                    {item.names}
+                  </Text>
+                </TouchableOpacity>
+              );
+            }}
+          />
+        </SafeAreaView>
+      ) : null}
+    </View>
+  );
+};
+
+export default DurationDropDown;
```

Comment:
Can we seperate the style in this file also like `src/components/ToDoComponent/SettingGoals/SettingGoalsComponents/Calendar.tsx`

Reply:
Sorry, I didn't get what you said here, could you please elaborate.

Reply:
Hi Jaya, why you are doing inline styling?

In this file `src/components/ToDoComponent/SettingGoals/SettingGoalsComponents/Calendar.tsx` I see there is seperate styling
https://github.com/Real-Dev-Squad/mobile-app/pull/169/files#diff-7c7e7d0ee01e44b976c3505b76d099ac6c9004e0e6ad501eed8b8895f2fa3e5fR34

Correct me If i am wrong because no idea of react-native

Reply:
Yeah, I got what you meant here and you are not wrong. I should have used a separate stylesheet instead of inline styling. Will make the necessary changes here. Thank you for noticing.

Reply:
Hello @shubhamsigdar1
I have created separate stylesheet. Could you please check?

Reply:
Hello, @shubhamsigdar1 Should I resolve this conversation?